### PR TITLE
Runtime: remove logger from bot component, exception handling on bot component load and pass IConfiguration to extensions

### PR DESCRIPTION
**Pull request by carlosscastro**

### CODE REVIEW

File: libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -81,55 +88,44 @@ internal static void AddBotRuntimeStorage(this IServiceCollection services)
...
-            
-            // Adapter settings so the default adapter is homogeneous with the configured adapters at the controller / registration level
-            services.AddSingleton(new AdapterSettings() { Route = defaultRoute, Enabled = true, Name = typeof(CoreBotAdapter).FullName });
 
-            // Adapter settings for configurable adapters. Runtime controllers pick up this config to get info on adapters and routes.
-            foreach (var adapterSetting in runtimeSettings.Adapters)
+            if (exceptions.Any())
             {
-                services.AddSingleton(adapterSetting);
+                throw new AggregateException("Encountered exceptions while loading bot components.", exceptions);
```
Comment (carlosscastro) [inline libraries/integration/Microsoft.Bot.Builder.Integration.Runtime/Extensions/ServiceCollectionExtensions.cs:118]: Moving to the pattern that asp.net core uses to load IHostingStartup enhancements

https://github.com/dotnet/aspnetcore/blob/80880d0ede83172aad8b48e8cd341b6a339ea8ff/src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs

**Review by tracyboehrer — APPROVED** [review]

File: libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorBotComponent.cs [hunk 2]
```diff
@@ -14,7 +14,7 @@ namespace Microsoft.Bot.Builder.AI.Orchestrator
     public class OrchestratorBotComponent : BotComponent
     {
         /// <inheritdoc/>
-        public override void ConfigureServices(IServiceCollection services, IConfiguration configuration, ILogger logger)
+        public override void ConfigureServices(IServiceCollection services, IConfiguration configuration)
```
Comment (carlosscastro) [inline libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorBotComponent.cs:17]: From https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-5.0

```
 Writing logs before completion of the DI container setup in the Startup.ConfigureServices method is not supported:

Logger injection into the Startup constructor is not supported.
Logger injection into the Startup.ConfigureServices method signature is not supported
The reason for this restriction is that logging depends on DI and on configuration, which in turns depends on DI. The DI container isn't set up until ConfigureServices finishes.
```

Comment (johnataylor) [reply libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorBotComponent.cs:17]: the constraint makes sense - you can add a component that depends on logging - but you wouldn't log the adding

**Review by johnataylor — APPROVED** [review]
Really loving this change!

**Pull request merged**